src/esp/sim/Simulator.cpp

```diff
@@ -1122,6 +1144,25 @@ agent::Agent::ptr Simulator::getAgent(const int agentId) {
   return agents_[agentId];
 }
 
+void Simulator::addSensorToObject(const int objectId,
+                                  const vec3f& position,
+                                  const vec3f& orientation,
+                                  const vec2i& resolution) {
+  esp::sensor::SensorSpec::ptr objectSensorSpec =
+      esp::sensor::SensorSpec::create_unique();
```

Comment:
Shouldn't this be `esp::sensor::SensorSpec::create()`?  Everywhere you reference these SensorSpecs you treat them as shared pointers, not unique ones.

Reply:
Otherwise, objectSensorSpec should be declared as SensorSpec::uptr

Reply:
Thanks John, that's fixed now and everything works :)